File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/TimerFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class TimerFileType extends LanguageFileType {
  public static final TimerFileType INSTANCE = new TimerFileType();

  private TimerFileType() {
    super(UnitFileLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Timer unit configuration for systemd";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "A unit configuration file whose name ends in \".timer\" encodes information about a timer controlled and supervised by "
           + "systemd, for timer-based activation.";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "timer";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return UnitFileIcon.FILE;
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/TimerFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class TimerFileType private constructor() : AbstractUnitFileType(UnitFileLanguage.INSTANCE) {
  override fun getName(): String {
    return "Timer unit configuration for systemd"
  }

  override fun getDescription(): String {
    return displayName
  }

  override fun getDefaultExtension(): String {
    return "timer"
  }

  override fun getIcon(): Icon? {
    return UnitFileIcon.TIMER
  }

  override fun getDisplayName(): @Nls String {
    return "Timer unit configuration (systemd)"
  }

  companion object {
    val INSTANCE = TimerFileType()
  }
}
